### PR TITLE
Backport PR #12208 on branch 3.3.x (Fix sdist editable install and add tests)

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -36,12 +36,12 @@ install_requires =
     tornado>=6.1.0
     jupyter_core
     jupyterlab_server~=2.10
-    jupyter_server~=1.4
+    jupyter_server~=1.8
```
Comment (fcollonval) [inline setup.cfg:39]: Let's not change the dependency for a patch version

```suggestion
    jupyter_server~=1.4
```

File: setup.cfg [hunk 2]
```diff
@@ -36,12 +36,12 @@ install_requires =
...
     nbclassic~=0.2
-    jinja2>=2.1
+    jinja2>=3.0.3
```
Comment (fcollonval) [inline setup.cfg:41]:
```suggestion
    jinja2>=2.1
```

Comment (blink1073) [reply setup.cfg:41]: We'll have to pin `markupsafe` then I believe

**Comment by blink1073** [comment]
Holy moly, we can now just re-run failed jobs only!

**Comment by blink1073** [comment]
jinja2 relies on an old API of setuptools, I think we need to just drop the min versions test in 3.3.x.   What do you think @fcollonval?

**Comment by fcollonval** [comment]
> jinja2 relies on an old API of setuptools, I think we need to just drop the min versions test in 3.3.x. What do you think @fcollonval?

:+1: @blink1073

**Pull request merged**